Source/Core/DolphinQt/Debugger/BranchWatchDialog.cpp

```diff
@@ -194,327 +187,370 @@ void BranchWatchProxyModel::SetInspected(const QModelIndex& index)
+
+  m_table_view->setModel(m_table_proxy);
+  m_table_view->setSortingEnabled(true);
+  m_table_view->sortByColumn(Column::Origin, Qt::AscendingOrder);
+  m_table_view->setSelectionMode(QAbstractItemView::ExtendedSelection);
+  m_table_view->setSelectionBehavior(QAbstractItemView::SelectRows);
+  m_table_view->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+  m_table_view->setContextMenuPolicy(Qt::CustomContextMenu);
+  m_table_view->setEditTriggers(QAbstractItemView::NoEditTriggers);
+  m_table_view->setCornerButtonEnabled(false);
+  m_table_view->verticalHeader()->hide();
+  m_table_view->setColumnWidth(Column::Instruction, 50);
+  m_table_view->setColumnWidth(Column::Condition, 50);
+  m_table_view->setColumnWidth(Column::OriginSymbol, 250);
+  m_table_view->setColumnWidth(Column::DestinSymbol, 250);
+  // The default column width (100 units) is fine for the rest.
+
+  auto* const horizontal_header = m_table_view->horizontalHeader();
+  horizontal_header->setContextMenuPolicy(Qt::CustomContextMenu);
+  horizontal_header->setStretchLastSection(true);
+  horizontal_header->setSectionsMovable(true);
+  horizontal_header->setFirstSectionMovable(true);
+
+  connect(m_table_view, &QTableView::clicked, this, &BranchWatchDialog::OnTableClicked);
+  connect(m_table_view, &QTableView::customContextMenuRequested, this,
+          &BranchWatchDialog::OnTableContextMenu);
+  connect(horizontal_header, &QHeaderView::customContextMenuRequested, this,
+          &BranchWatchDialog::OnTableHeaderContextMenu);
+  connect(new QShortcut(QKeySequence(Qt::Key_Delete), this), &QShortcut::activated, this,
+          &BranchWatchDialog::OnTableDeleteKeypress);
+
+  // Status Bar
+  m_status_bar = new QStatusBar(this);
+  m_status_bar->setSizeGripEnabled(false);
+
+  // Controls Toolbar
+  m_control_toolbar = new QToolBar(this);
+  {
     // Tool Controls
-    m_control_toolbar->addWidget([this]() {
-      auto* const layout = new QGridLayout;
+    m_btn_start_pause = new QPushButton(tr("Start Branch Watch"), this);
+    connect(m_btn_start_pause, &QPushButton::toggled, this, &BranchWatchDialog::OnStartPause);
+    m_btn_start_pause->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Expanding);
+    m_btn_start_pause->setCheckable(true);
+
+    m_btn_clear_watch = new QPushButton(tr("Clear Branch Watch"), this);
+    connect(m_btn_clear_watch, &QPushButton::clicked, this, &BranchWatchDialog::OnClearBranchWatch);
+    m_btn_clear_watch->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Expanding);
+
+    m_btn_path_was_taken = new QPushButton(tr("Code Path Was Taken"), this);
+    connect(m_btn_path_was_taken, &QPushButton::clicked, this,
+            &BranchWatchDialog::OnCodePathWasTaken);
+    m_btn_path_was_taken->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Expanding);
+
+    m_btn_path_not_taken = new QPushButton(tr("Code Path Not Taken"), this);
+    connect(m_btn_path_not_taken, &QPushButton::clicked, this,
```

Comment:
I think these removed `layout->addWidget` lines are fine and can be kept. IIRC, these `QPushButton` will be reparent'd when passed to the layout's `addWidget` method as mentioned here: https://doc.qt.io/qt-6/layout.html

> **Tips for Using Layouts**
When you use a layout, you do not need to pass a parent when constructing the child widgets. The layout will automatically reparent the widgets (using [QWidget::setParent](https://doc.qt.io/qt-6/qwidget.html#setParent)()) so that they are children of the widget on which the layout is installed.

You might need to set a parent to the layout before adding widgets to it, though. So feel free to keep your change if you want to defer the layout construction, you just don't need to pass `this` as a parent to the layout's child widgets' constructor.

Reply:
I was working under an assumption that was missing this detail. The reason I was parenting everything to `this` was because it seemed the only consistent way to ensure child widgets were destructed properly. I may need to rethink some priorities in the constructor now...